src/cli/Generate.cpp

```diff
@@ -120,9 +127,15 @@ QSharedPointer<PasswordGenerator> Generate::createGenerator(QSharedPointer<QComm
 
     // The default charset will be used if no explicit class
     // option was set.
-    passwordGenerator->setCharClasses(classes);
-    passwordGenerator->setFlags(flags);
-    passwordGenerator->setExcludedChars(parser->value(Generate::ExcludeCharsOption));
+    if (flags != 0x0) {
+        passwordGenerator->setFlags(flags);
+    }
+    QString customCharacterSet = parser->value(Generate::CustomCharacterSetOption);
+    if (classes != 0x0 || !customCharacterSet.isNull()) {
```

Comment:
I feel like the handling of the 0x0 variable passed in should be done by the password generator class. By explicitly bypassing the calls here we are implicitly manipulating how the password generator works. To me, 0x0 should mean "Use the defaults" to the password generator.

Reply:
I've added the reset method to the PasswordGenerator class to have a more explicit interface. When we use 0x0 to reset values it won't be possible to not set any flag at all, or, perhaps more importantly, use only the custom character class. 

We could go back to an interface with a setOptions(flags, active_classes, excludedCharacters, customCharacters) and then handle such cases accordingly (i.e., reset active_classes to default only when customCharacters is not set), but i feel like this is somewhat of an antipattern as it hides a lot of logic.